### PR TITLE
[PURCHASE-2005] Fix issue with details panel

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **Bugfix**

This PR resolves: **[[PURCHASE-2005](https://artsyproduct.atlassian.net/browse/PURCHASE-2005)]**

### Description

- Details panel should remain open when switching between conversations
- Hide Details panel on user screen resize for small screens

### Screenshots

ezgif-2-78f6fac974e6

### CODE REVIEW

File: src/v2/Apps/Conversation/routes.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const conversationRoutes: RouteConfig[] = [
   {
     path: "/user/conversations/:conversationID",
     displayFullPage: true,
-    getComponent: () => loadable(() => import("./Routes/Conversation")),
+    Component: loadable(() => import("./Routes/Conversation")),
```
Comment (damassi) [inline src/v2/Apps/Conversation/routes.tsx:29]: Why change to `Component`?

Comment (sepans) [reply src/v2/Apps/Conversation/routes.tsx:29]: I leave it to @MounirDhahri to explain the details but as far as I understood this change fixed the bug of `showDetails` being reset when navigating between conversations. (cc: @zephraph since you have set this up)

File: src/v2/Apps/Conversation/Components/Details.tsx [hunk 2]
```diff
@@ -80,6 +110,19 @@ export const Details: FC<DetailsProps> = ({ conversation, ...props }) => {
...
+    // For big screens
+    if (width > parseInt(breakpoints.xs, 10)) {
+      if (showDetails) {
+        return "376px"
+      }
+      return ["376px", "376px", "376px", "376px", "0"]
+    }
+
+    // For small screens
+    return "100%"
```
Comment (sepans) [inline src/v2/Apps/Conversation/Components/Details.tsx:123]: Cool 💯  It looks much better to me but good to double-check with Will (if you haven't done already) if detail box has to be full width on mobile.

File: src/v2/Apps/Conversation/Routes/Conversation/index.tsx [hunk 3]
```diff
@@ -30,7 +30,6 @@ const ConstrainedHeightFlex = styled(Flex)`
     height: calc(100vh - 55px);
   `}
   & > * {
-    overflow-y: scroll;
```
Comment (sepans) [inline src/v2/Apps/Conversation/Routes/Conversation/index.tsx:33]: Didn't understand 👆. was this change intentional?

File: src/v2/Apps/Conversation/routes.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/v2/Apps/Conversation/routes.tsx:29]: When I checked the profiler I found out that `showDetails` is being reset because the conversation is getting mounted again whenever the user navigates to a new conversation, I first tried to wrap it with a memo to avoid that but found out that `getComponent` was the issue since React wasn't able to tell if that it's the same component or not to avoid rendering it.

File: src/v2/Apps/Conversation/Components/Details.tsx [hunk 4]
```diff
@@ -51,9 +55,35 @@ const DetailsContainer = styled(Flex)<{ opacity?: 0 | 1; transform?: string }>`
...
+        initialWidth > parseInt(breakpoints.xs, 10) &&
+        showDetails
+      ) {
+        setShowDetails(false)
+      }
+    })
+    window.addEventListener("resize", listenForResize)
+    return () => window.removeEventListener("resize", listenForResize)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [setShowDetails, showDetails])
```
[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/v2/Apps/Conversation/Components/Details.tsx:85]: yes sure

File: src/v2/Apps/Conversation/Routes/Conversation/index.tsx (hunk 3 above)
Comment (MounirDhahri) [reply src/v2/Apps/Conversation/Routes/Conversation/index.tsx:33]: oh no

File: src/v2/Apps/Conversation/routes.tsx (hunk 1 above)
Comment (sepans) [reply src/v2/Apps/Conversation/routes.tsx:29]: 🧠
